parsl/executors/flux/executor.py

```diff
@@ -0,0 +1,247 @@
+    flux_handle_args: collections.abc.Sequence
+        Positional arguments to ``flux.Flux()`` instance, if any.
+    flux_handle_kwargs: collections.abc.Mapping
+        Keyword arguments to pass to ``flux.Flux()`` instance, if any.
+        For instance, the ``url`` argument to ``flux.Flux`` can be used to connect to a
+        remote Flux instance at the given URL.
+    """
+
+    def __init__(
+        self,
+        managed: bool = True,
+        working_dir: Optional[str] = "flux_workdir",
+        label: str = "FluxExecutor",
+        flux_handle_args: Sequence = (),
+        flux_handle_kwargs: Mapping = {},
+    ):
+        super().__init__()
+        self.label = label
+        self.working_dir = os.path.abspath(working_dir) if working_dir else os.getcwd()
+        self.managed = managed
+        self._flux_executor = None
+        self.flux_handle_args = flux_handle_args
+        self.flux_handle_kwargs = flux_handle_kwargs
+        self._provider = LocalProvider()
+
+    def start(self):
+        """Called when DFK starts the executor when the config is loaded.
+
+        Raises
+        ------
+        OptionalModuleMissing
+            If ``flux`` package cannot be imported.
+        """
+        if not _FLUX_AVAIL:
+            raise OptionalModuleMissing(
+                "flux", "Cannot initialize FluxExecutor without flux module"
+            )
+        self._flux_executor = flux.job.FluxExecutor(
+            handle_args=self.flux_handle_args, handle_kwargs=self.flux_handle_kwargs
+        )
+        os.makedirs(self.working_dir, exist_ok=True)
+
+    def shutdown(self, wait=True):
+        """Shut down the executor, causing further calls to ``submit`` to fail.
+
+        Parameters
+        ----------
+        wait
+            If ``True``, do not return until all submitted Futures are done.
+        """
+        self._flux_executor.shutdown(wait=wait)
+
+    def submit(self, func: Callable, resource_spec: Mapping, *args, **kwargs):
+        """Wrap a callable in a Flux job and submit it to Flux.
+
+        Parameters
+        ----------
+        func: callable
+            The callable to submit as a job to Flux
+        resource_spec: collections.abc.Mapping
+            A mapping defining the resources to allocate to the Flux job.
+            Only the following keys are checked for:
+            * num_tasks: the number of tasks to launch (MPI ranks for an MPI job),
+              default 1
+            * cores_per_task: cores per task, default 1
+            * gpus_per_task: gpus per task, default 1
+            * num_nodes: if > 0, evenly distribute the allocated cores/gpus
+              across the given number of nodes. Does *not* give the job exclusive
+              access to those nodes; this option only affects distribution.
+        *args:
+            positional arguments for the callable
+        **kwargs:
+            keyword arguments for the callable
+        """
+        tid = str(uuid.uuid4())
```

Comment:
I wonder if it might be easier to use and debug if the task IDs were 0, 1, 2, 3, ..., and every executor instance was given a unique workdir (so, unique workdirs instead of unique file names). As it stands, with all the files produced, it can be very hard to remember which app/future goes with which UUID, because the UUIDs are so long. There would just be less to memorize.

Reply:
You could use an atomic counter, and it's probably nicer than UUIDs.

Reply:
If you're grepping through a log file, there's no way a UUID is anything else but what it's supposed to be. If, on the other hand, you're grepping for `1`, you might get a few more extra hits.